### ISSUE TITLE
junit xml reports are not properly formatted on failure when coloring is enabled

### ISSUE THREAD

**Issue by RafalSkorka** [webdriverio/webdriverio#6358]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 6.12..1
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 14.15.4
 - **NPM version:** 6.14.10
 - **Browser name and version:** NA
 - **Platform name and version:** Windows 10
 - **Additional wdio packages used (if applicable):** @wdio/junit reporter

**Config of WebdriverIO**
Configuration generated with `npx wdio config -y` and then added junit reporter:
```
    reporters: ['spec',
        ['junit', {
            outputDir: './reports',
            outputFileFormat: function (options) { // optional
                return `results-${options.cid}.${options.capabilities}.xml`
            }
        }]],
```

**Describe the bug**
When terminal supports coloring the error message in junit xml file contains invalid characters used for output coloring like this:
```
<error message="Expect $(`#flash`) not to be existing&#xA;&#xA;Expected [not]: [32m&quot;[7mnot [27mexisting&quot;[39m&#xA;Received      : [31m&quot;existing&quot;[39m"/>
```
Such XML is then rejected by junit reporter in Jenkins as it contains 0x1b character.

**To Reproduce**
Go with the "Getting started" procedure and add junit reporter. Then modify any test to make it fail eg. I added `not` here:
```
        expect(SecurePage.flashAlert).not.toBeExisting();
```

**Expected behavior**
The xml generated by junit reporter should be valid.

**Log**
![obraz](https://user-images.githubusercontent.com/3476094/106019623-232d6980-60c3-11eb-9069-fd584f0d791b.png)

**Additional context**
It is reproducible on Windows, but I am actually hitting this issue when used in Jenkins which then reports such test as
```
results-0-0.chrome.xml.[failed-to-read]

Failed to read test report file /redacted/e2e/reports/results-0-0.chrome.xml
org.dom4j.DocumentException: Error on line 878 of document  : An invalid XML character (Unicode: 0x1b) was found in the value of attribute "message" and element is "error".
	at org.dom4j.io.SAXReader.read(SAXReader.java:511)
	at org.dom4j.io.SAXReader.read(SAXReader.java:392)
	at hudson.tasks.junit.SuiteResult.parse(SuiteResult.java:178)
	at hudson.tasks.junit.TestResult.parse(TestResult.java:378)
	at hudson.tasks.junit.TestResult.parsePossiblyEmpty(TestResult.java:308)
	at hudson.tasks.junit.TestResult.parse(TestResult.java:224)
	at hudson.tasks.junit.TestResult.parse(TestResult.java:196)
	at hudson.tasks.junit.TestResult.<init>(TestResult.java:151)
	at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:144)
	at hudson.FilePath.act(FilePath.java:1161)
	at hudson.FilePath.act(FilePath.java:1144)
	at hudson.tasks.junit.JUnitParser.parseResult(JUnitParser.java:107)
	at hudson.tasks.junit.JUnitResultArchiver.parse(JUnitResultArchiver.java:149)
	at hudson.tasks.junit.JUnitResultArchiver.parseAndSummarize(JUnitResultArchiver.java:243)
	at hudson.tasks.junit.pipeline.JUnitResultsStepExecution.run(JUnitResultsStepExecution.java:63)
	at hudson.tasks.junit.pipeline.JUnitResultsStepExecution.run(JUnitResultsStepExecution.java:29)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

As a workaround I could probably disable the ansiColor plugin, but then the test output will not look nice anymore :)

**Comment by christian-bromann** [comment]
This indeed seems like a bug. The XML report contains these characters that need to be stripped away:

```xml
<error message="Expect $(`#flash`) to have text containing&#xA;&#xA;[32m- Expected  - 1[39m&#xA;[31m+ Received  + 2[39m&#xA;&#xA;[32m- You logged into a secure area![7m WRONG![27m[39m&#xA;[31m+ You logged into a secure area![39m&#xA;[31m+ ×[39m"/>
```

at least for the error message. Contributions to this are very welcome!

**Comment by RafalSkorka** [comment]
I've raised an issue to the [junit-report-builder](https://github.com/davidparsson/junit-report-builder/issues/20) library, so when it is fixed, it would be enough to bump the version number here.

**Comment by davidparsson** [comment]
The junit-report-builder issue [mentioned above](https://github.com/webdriverio/webdriverio/issues/6358#issuecomment-777735610) is fixed and released in [3.0.0](https://www.npmjs.com/package/junit-report-builder/v/3.0.0), which also drops support for node.js 4 and 6.

**Comment by christian-bromann** [comment]
fixed in #6430

**Comment by harveyincinerator1** [comment]
@christian-bromann https://breakthrough363.github.io/separated/
